### PR TITLE
Dying and restart implementation

### PR DESCRIPTION
**Pull request by 1ukasz**
Imo all of the game features works properly. Of course refactor/cleaning would be nice but I want to try to convert it to TS anyway, and I just wanted to be the first (am I?). 😄

### CODE REVIEW

File: js/game.js [hunk 1]
```diff
@@ -99,48 +104,91 @@ export default class Game {
...
-        `-${this.tamagotchi.states.playing}`,
-        this.tamagotchi.states.playing.toUpperCase()
-      );
-      this.tamagotchi.fun.value += 2;
-      this.displayTimeout(
-        this.tamagotchi.displayFun,
-        this.tamagotchi.fun.element
-      );
+      if (!this.isPlaying) {
+        this.isSleeping = false;
```
Comment (Sumick) [inline js/game.js]: **nitpick:** I forgot to mention it in the previous PR, but we should definitely split these methods independently, eg:
```handlePlay, handleSleep, etc.```. Currently it is too big. I would suggest to use `Abilities` class that you have currently unused.

**Review by Sumick — CHANGES_REQUESTED** [review]
Nice! I caught two things to fix:

- [x] Let's split your Tamagotchi abilities (sleep, eat, play) into separate methods (it is basically a part of your previous PR, but I forgot to mention about it there)
- [x] Let's fix you parameters recovery. It should recover above base value. I will update task description, so it's more clear. (screen)

Zrzut ekranu 2023-03-6 o 14 43 25

File: js/game.js (hunk 1 above)
Comment (1ukasz) [reply js/game.js]: Yeah, I've been trying to move few methods to Abilities.js by myself even earlier but I failed. Too many params from tamagotchi mixed with params from Game module, so even if imo I've been moving methods properly the code still didn't want to work as it works now from game.js 😉 so I left it here.

**Comment by 1ukasz** [comment]
> * Let's fix you parameters recovery. It should recover above base value. I will update task description, so it's more clear. (screen)

Nie wiem, czy dobrze zrozumiałem ale zablokowałem parametry, żeby nie przekraczały tej bazowej wartości 10. Mam nadzieję, że o to Ci chodziło  😉

**Review by Sumick — APPROVED** [review]
Looks great! Amazing job 🙌

PS. Yes, you are first 👑

**Review by Sumick — CHANGES_REQUESTED** [review]
Oh - I noticed one more thing! Other parameters should not stop decreasing when action is on. Example: when Tamago is eating, his energy should still be decreasing.

**Review by Sumick — APPROVED** [review]
Looks good! Congrats 🙌

**Pull request merged**